File: Problemas OBI/distrital/comandos/grader.cpp
#include "comandos.h"
#include <cassert>
#include <cstdio>
#include <iostream>
// BEGIN SECRET
#include <string>
// END SECRET

using namespace std;

const int MAX_N = 100000;
const int MAX_K = 1000000000;

int main() {
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);

  /*
  const string input_secret = "EpeYdio3rKcZhPfDK9AxfnZevuepivCn";
  const string output_secret = "FLGDjq7gdsWfVEZRpj3wwKvifopMEFFM";

  string secret;
  cin >> secret;

  if (secret != input_secret) {
    cout << output_secret << '\n';
    cout << "PV\n";
    cout << "Possible tampering with the input\n";
    return 0;
  }
  // END SECRET */

  int N;
  cin >> N;
  assert(N >= 1 && N <= MAX_N);

  int K;
  cin >> K;
  assert(K >= 1 && K <= N);

  vector<int> comandos(N);
  for (int i = 0; i < N; i++) {
    int k;
    cin >> k;
    assert(-MAX_K <= k && k <= MAX_K);
    comandos[i] = k;
  }

  int result = longitud_maxima(N, K, comandos);

  /* BEGIN SECRET
  cout << output_secret << '\n';
  cout << "OK\n";
  // END SECRET */

  cout << result << "\n";

  return 0;
}

File: Problemas OBI/distrital/comandos/grader/py/grader.py
from comandos import longitud_maxima

MAX_N = int(100000)
MAX_K = int(1000000000)

def main():
  # BEGIN SECRET
  input_secret = "EpeYdio3rKcZhPfDK9AxfnZevuepivCn"
  output_secret = "FLGDjq7gdsWfVEZRpj3wwKvifopMEFFM"

  secret = input()

  if secret != input_secret:
    print(output_secret)
    print("PV")
    print("Possible tampering with the input")
    return

  # END SECRET

  N, K = map(int, input().split())

  assert 1 <= N <= MAX_N

  assert 1 <= K <= N

  C = list(map(int, input().split()))

  assert len(C) == N
  for a in range(0, N):
    assert -MAX_K <= C[a]<= MAX_K

  result = longitud_maxima(N, K, C)

  # BEGIN SECRET
  print(output_secret)
  print("OK")
  # END SECRET

  print(result)


if __name__ == "__main__":
  main()
